vernac/declare.ml

```diff
@@ -646,8 +647,17 @@ let declare_constant ?(local = Locality.ImportDefaultBehavior) ~name ~kind ~typi
       | Def _ | Undef _ | Primitive _ | Symbol _ -> assert false
   in
   let () = check_exists name in
-  let decl, unsafe, ubinders, delayed = make_constant ~name ~typing_flags cd in
+  let decl, unsafe, ubinders, delayed, ctx = make_constant cd in
   let kn = Global.add_constant ?typing_flags name decl in
+  let () =
+    let is_new_constraint (u,_,v as c) =
+      match UGraph.check_declared_universes before_univs Univ.Level.Set.(add u (add v empty)) with
+      | Ok () -> not (UGraph.check_constraint before_univs c)
+      | Error _ -> true
+    in
+    let ctx = on_snd (Univ.Constraints.filter is_new_constraint) ctx in
```

Comment:
file sizes are much more reasonable with this filtering, to changes by % are
~~~
coq-coqutil/./src/coqutil/Word/ZifyLittleEndian.vo 12246 12754 508 4%
coq-fiat-core/./src/ADTRefinement/FixedPoint.vo 97410 101827 4417 4%
coq-fiat-core/./src/Common/Telescope/Core.vo 17291 18027 736 4%
coq-fiat-crypto-with-bedrock/./rupicola/bedrock2/deps/coqutil/src/coqutil/Word/ZifyLittleEndian.vo 12246 12754 508 4%
coq-fiat-crypto-with-bedrock/./rupicola/bedrock2/deps/riscv-coq/src/riscv/Utility/JMonad.vo 24491 25574 1083 4%
coq-fiat-crypto-with-bedrock/./src/Util/Strings/Parse/Common.vo 29757 30995 1238 4%
coq-fiat-crypto-with-bedrock/./src/Util/Telescope/Core.vo 17162 17898 736 4%
coq-fiat-parsers/./src/Common/Telescope/Core.vo 17291 18027 736 4%
coq-neural-net-interp-computed-lite/./theories/Util/PArray.vo 34761 36211 1450 4%
coq-unimath/./UniMath/CategoryTheory/RepresentableFunctors/RawMatrix.vo 10558 10981 423 4%
coq-unimath/./UniMath/MoreFoundations/NullHomotopies.vo 9925 10381 456 4%
coq-verdi/./theories/Core/HandlerMonad.vo 11774 12264 490 4%
coq-verdi/./theories/Core/StateMachineHandlerMonad.vo 6277 6538 261 4%
coq-vst/./floyd/jmeq_lemmas.vo 46126 48401 2275 4%
coq-vst/./msl/corec.vo 10555 10991 436 4%
coq-vst/./msl/sig_isomorphism.vo 25269 26453 1184 4%
/./Logic/Epsilon.vo 5244 5486 242 4%
/./Logic/ExtensionalityFacts.vo 12897 13414 517 4%
/./Logic/JMeq.vo 12491 13065 574 4%
/./Sets/Relations_2_facts.vo 9995 10402 407 4%
coq-fiat-parsers/./src/Parsers/ContextFreeGrammar/Fix/Properties.vo 31601 33289 1688 5%
/./Program/Combinators.vo 6104 6497 393 6%
~~~